zk/stages/test_utils.go

```diff
@@ -43,9 +40,24 @@ func (c *TestDatastreamClient) ReadAllEntriesToChannel() error {
 		c.entriesChan <- &c.gerUpdates[i]
 	}
 
+	c.entriesChan <- nil // needed to stop processing
+
+	for {
+		if c.stopReadingToChannel.Load() {
+			break
+		}
+	}
+
 	return nil
 }
 
+func (c *TestDatastreamClient) RenewEntryChannel() {
```

Comment:
empty ?

Reply:
Yes, it is not needed for the tests, but needed for the interface.